fastify.js

```diff
@@ -271,15 +271,11 @@ function build (options) {
   }
 
   function _handleTrustProxy (req) {
-    req.ip = proxyAddr(req, proxyFn)
-    req.ips = proxyAddr.all(req, proxyFn)
-    if (req.ip !== undefined) {
-      req.hostname = req.headers['x-forwarded-host']
-    }
+    return { ip: proxyAddr(req, proxyFn), ips: proxyAddr.all(req, proxyFn), get hostname () { return this.ip !== undefined ? req.headers['x-forwarded-host'] : req.hostname } }
```

Comment:
`hostname` doesn't need to be a getter. It can be a regular property.

Also, `req.hostname` isn't part of Node's [`IncomingMessage`](https://nodejs.org/dist/latest-v11.x/docs/api/http.html#http_class_http_incomingmessage) object, so `: req.hostname` should be replaced with `: undefined` here.

Reply:
I did try it `hostname: this.ip !== undefined ? req.headers['x-forwarded-host'] : undefined ` based on your previous comment, but trust-proxy.test.js tests failed
```
test/trust-proxy.test.js ............................. 5/17 30s
  trust proxy
  not ok Cannot read property 'ip' of undefined
    stack: |
      _handleTrustProxy (fastify.js:274:92)
      Object.routeHandler [as handler] (fastify.js:297:28)
      Router.lookup (node_modules/find-my-way/index.js:338:14)
    at:
      line: 274
      column: 92
      file: fastify.js
      function: _handleTrustProxy
    type: TypeError
    test: trust proxy
    source: >
      return { ip: proxyAddr(req, proxyFn), ips: proxyAddr.all(req, proxyFn),
      hostname: this.ip !== undefined ? req.headers['x-forwarded-host'] : undefined
      }
```

Reply:
Oh I see what happened. Looks like `ip` should be stored as a variable and then you can replace `this.ip` with just `ip`.

Reply:
resolved e446c9f